### ISSUE TITLE
structure not intuitive how to query across folders (via improved structure filter ui/ux)

### ISSUE THREAD

**Issue by davilima6** [plone/mockup#671]
# Implementation

Trying  to solve #381 I propose:
1. **Initial value for Title criteria:** When user inputs something in Live Filter and then clicks the `Query` button, Plone should open the Query popover with initial `Title` criteria filled with that input. This makes it explicit why and how the current listing is filtered already and why user won't see all results even if he/she clears all the popover filters.
2. **Different buttons for Live filter and Query popover :** using `glyphicon-search` suggests users to think that button is needed to be pressed after providing input in the text field, which 1) isn't true, as the filtering happens by keystroke and 2) leads to unexpected ui behavior as a popover opens (instead of searching/filtering/querying). So I suggest:
   1. Update placeholder from "Filter" to "Filter items" to make label paralel to search viewlet's placeholder
   2. Add a button with `.glyphicon-filter` or `.glyphicon-option-horizontal` for the Advanced Query popover 
   3. Move `.glyphicon-search` inside and to the right of the input field, to reinforce this is a search input widget while still keeping it visually simple with only one explicit button (besides the search one would still work when js is disabled so no a11n loss)
   4. Add styles to new Query button to indicate query state, i.e. whether any criteria other than Title (which is already visible in Live filter) is set. For the relative path criteria, state would be empty when different than current folder's path.
3. **In-page live filter** should return items **_containing_ querystring**, instead of **only starting with it.**, as in  plone/plone.app.users#54.
4. Add a `in current folder` checkbox (ticked by default) beneath live filter. Besides, add `Location` criteria by default on the `More filters` popover. When checkbox is on, `Location` is set to `Current (./)`. When it's off, `Location` value is default, that is, `Custom`, which provides a nice navigation widget initially set at `entire site` level. This way users gets expected behavior (wider search) and at the same time the chance to refine their query.
# Risks
- Too many buttons/clutter?

**Comment by davilima6** [comment]
If there's consensus I can work on the tasks above.

@vangheem @djay @jensens @thet what do u think?

**Comment by jensens** [comment]
All sounds good to me. 

Only thing I'd like to see additional is a kind of "Reset Query" feature (button, broom-icon, ...).

**Comment by djay** [comment]
Cool but #381 had two options, which one are you implementing?

 Perhaps you need mockup to show how all these ideas are going to look. Its a bit  hard to follow.

 It would also be good if you copied the original title and description from #381 so its clear what's the main problem you are trying to solve. Issues like ux enhancements are confusing. @plone/ux-team.

**Comment by davilima6** [comment]
@djay I don't think #381 options are exclusive, do you? I'd go with both. Besides I don't think I should copy any of it inside of this issue: IMO they relate but are separate so just a reference should be enough.

Re: the problems being solved, would you have any explicit doubt to ask? I think I either stated the reasoning for each of the changes proposed, or it is clear by itself, like item 3.

**Comment by djay** [comment]
@davilima6 I've changed your issue to make it explicit the problem you are trying to fix, and the things you seem to be proposing to do from #381. Please fix it if its not correct. I still think it would be clearer to show a quick mockup of everything you propose working together as I'm not sure I understand it all but up to you.

**Comment by jensens** [comment]
+1 for a quick mock or scribble, even if I think I do understand what you want to implement.

**Comment by davilima6** [comment]
Here's a small mockup for items #1 and #2: https://dl.dropboxusercontent.com/u/5766075/screencasts/live-filter.mp4

Please picture that cleaning or updating the popover's Title criteria would clear the in-page textual filter automatically. I didn't mock that. Also I didn't mock #2 item (iv). This could go in another iteration, I imagine changing filter icon's color or adding a discreet dot to indicate there are active filters inside the popover.

I think #3 is self explanatory, also it's the same suggestion as in plone/plone.app.users#54

Finally I was going to mock #4 but got into some errors for which I'll open a new issue

**Comment by davilima6** [comment]
I just noticed the in-page live filter operates on SearchableText so please picture that instead of the Title criteria when I open the Query popover in the video.

**Comment by davilima6** [comment]
So, @jensens, @djay, could you manage to get a better idea or would you have any other doubt about my proposal? May I proceed with implementing?

**Comment by djay** [comment]
ok. I get num 2 now. Looks great. I think the "more" button works really well.

Perhaps at the same time we can improve the title "Query" as this is a little meaningless. Perhaps it should be "filter", "search" or "advanced search"? ... although that might be confusing if the path is not set and therefore showing the whole site. Perhaps this could be made more explicit by having a path = / to show its searching the whole site? (unless it has another path set of course)

num 4 is a good idea too.

**Comment by davilima6** [comment]
Agree. I'd suggest "More filters" as popover title so it completely reflects the button. What do you think?

Re: your other doubt, I finally updated what you wrote for item 4. Tell me if that helps/make sense etc. Thanks!

**Comment by djay** [comment]
@davilima6 by item 4, do you mean "Beneath the search box there is a "current folder" tick box (ticked by default). If you click query with this ticked it will include an explicit relative path criteria in the search box"
I agree with this. I was suggesting in addition, if "current folder" is not ticked, when you click on the more filters you get an explicit absolute path criteria of "/", as that makes it clear it will search the whole site.

**Comment by davilima6** [comment]
Yes, that's what I'm referring to, however I updated the issue summary, so please refresh.

**Comment by djay** [comment]
Sorry. Obviously github live updates new comments but not edits. That looks great.

**Comment by jensens** [comment]
Like! Yes please proceed!

**Comment by fgrcon** [comment]
any progress here?
btw. querying by ``Short Name (id)``  is not very very useful right now:  can only choose ``is`` instead of `contains` .

**Comment by thet** [comment]
Mostly fixed by: https://github.com/plone/mockup/pull/937
Some items maybe not. If so, please open another issue.
